### DISCUSSION TITLE
Is there (or will be a official) opentelemetry provider library for bun?

### DISCUSSION THREAD

**Discussion by costiss** [oven-sh/bun#7185] [Q&A] [answered]
Is there (or will be) any opentelemetry supporting library for bun?

All the NodeJS ones that I've tried did not work

**Comment by Jarred-Sumner** [comment, marked as answer]
Maybe

I do think a telemetry API is something that a runtime should provide

**Reply by Costiss** [reply]
Thanks for the answer @Jarred-Sumner 

That's great!

I believe that the challenges for integrating observability represents a significant role on new projects that are not adopting Bun, although they wanted to.

**Reply by admmasters** [reply]
Yeah we can't go to production with anything that doesn't have telemetry.

**Reply by opensourceinterfaces292** [reply]
Agree!

**Reply by schettn** [reply]
Any idea on when this will be supported?

**Reply by vladkrasn** [reply]
Would be real nice

**Reply by SpartanDavie** [reply]
Is this still being considered? I've been watching the amazing progress of Bun but this has been the one thing stopping me doing a project with it.

**Comment by jkone27** [comment]
seems deno2.2 just added native open telemetry support

**Comment by aarvay** [comment]
Thanks, native OpenTelemetry (traces, metrics, W3C context/OTLP) is a critical blocker for production adoption and is forcing teams to avoid Bun or build fragile workarounds. Is there a concrete plan or timeline to prioritize this? Happy to help test or contribute.

Enabling telemetry would unlock a lot more real-world adoption.

**Reply by eduvhc** [reply]
Indeed, we were considering Bun, but found this during the analysis phase.

**Reply by SpartanDavie** [reply]
Same

**Reply by perezd** [reply]
Jared said this to me: https://bsky.app/profile/jarredsumner.com/post/3mmq3rhwxvs2f

**Comment by Drejerdk** [comment]
I would very much appreciate this. Been watching this for more than a year now, hoping for a native solution. 🙂

**Reply by zessu** [reply]
I have had no problems integrating open telemetry in any bun projects. why should this be native? okay but if Bun supports native modules for s3 maybe why not but again why should they do this?

**Reply by perezd** [reply]
What libraries did you end up using? Curious if there are more than the standard package: https://github.com/open-telemetry/opentelemetry-js/tree/main

**Reply by zessu** [reply]
> What libraries did you end up using? Curious if there are more than the standard package: https://github.com/open-telemetry/opentelemetry-js/tree/main

hey sorry just seeing this . yes basically used the same set up https://opentelemetry.io/docs/languages/js/getting-started/nodejs/

**Comment by MichaelTendoSsemwanga** [comment]
+1

**Comment by Gumimaco** [comment]
Any update on this?

**Comment by acoBOYZ** [comment]
Up

**Comment by sebastianmauro** [comment]
Any update on this? maybe a workaround?
